script/gen_requirements_all.py

```diff
@@ -295,34 +308,44 @@ def process_requirements(errors, module_requirements, package, reqs):
 
 def generate_requirements_list(reqs):
     """Generate a pip file based on requirements."""
+    separate = True
+    if isinstance(reqs, list):
+        reqs = dict.fromkeys(reqs, ())
+        separate = False
     output = []
-    for pkg, requirements in sorted(reqs.items(), key=lambda item: item[0]):
-        for req in sorted(requirements):
-            output.append(f"\n# {req}")
+    for pkg, requirements in sorted(
+        reqs.items(), key=lambda item: get_package_sort_key(item[0])
+    ):
+        if separate:
+            output.append("")
+        for req in sorted(requirements, key=get_package_sort_key):
+            output.append(f"# {req}")
 
         if comment_requirement(pkg):
-            output.append(f"\n# {pkg}\n")
+            output.append(f"# {pkg}")
         else:
-            output.append(f"\n{pkg}\n")
-    return "".join(output)
+            output.append(pkg)
+    output.append("")  # Add empty line at end
+    return "\n".join(output)
 
 
 def requirements_output(reqs):
     """Generate output for requirements."""
     output = [
+        AUTOGENERATED_ADMONITION,
         "-c homeassistant/package_constraints.txt\n",
         "\n",
         "# Home Assistant Core\n",
     ]
-    output.append("\n".join(core_requirements()))
-    output.append("\n")
+    output.append(generate_requirements_list(core_requirements()))
```

Comment:
If you do the sorting here inside `requirements_output`, then you can remove the added complexity in `generate_requirements_list`.

```suggestion
    output.append("\n".join(sorted(core_requirements(), key=get_package_sort_key)))
    output.append("\n")
```

Reply:
No – `generate_requirements_list` is called by all `requirements_*` functions, and I want all of them to have the exact same sorting behavior without needing to touch all of them.

Reply:
Please leave open for other reviewers to comment. I really don't see any advantage to your added complexity.

Reply:
I agree with @epenet on this one.